### PR TITLE
Handle `OneHot`'s `depth` value during shape inference

### PR DESCRIPTION
**Pull request by maekawatoshiki**
### Description

- Even if the `depth` value of `OneHot` is specified as an initializer, the value is ignored during shape inference currently.
- This is a correct shape inference from my understanding, but let me if this is wrong.

### CODE REVIEW

File: onnx/defs/tensor/defs.cc
Comment (maekawatoshiki) [inline onnx/defs/tensor/defs.cc:2904]: It might not be int64. Any suggestion to easily handle int8/16/32/64 data?

Comment (gramalingam) [reply onnx/defs/tensor/defs.cc:2904]: Check the type and handle it case-by-case: eg., like [here](https://github.com/onnx/onnx/blob/d7e2b813c1a26cdb8bcb6c4547cb5e6ffeb9b49b/onnx/defs/sequence/defs.cc#L347) ? I guess it would be nice to create an utility function others can also use.

Comment (maekawatoshiki) [reply onnx/defs/tensor/defs.cc]: I realized that [onnxruntime is now supporting only float/int32/int64 for depth](https://github.com/microsoft/onnxruntime/blob/cd56ea4a74ee41c040899d702667d2c86bee4ef0/onnxruntime/core/providers/cpu/tensor/onehot.cc#L59-L69).

**Comment by maekawatoshiki** [comment]
How's the review process going?

**Comment by justinchuby** [comment]
LGTM. I will leave this to @SIG-operators-approvers reviewers

**Review by gramalingam — APPROVED** [review]

**Pull request merged**
